tecken/tests/test_upload.py

```diff
@@ -469,70 +254,55 @@ def mock_api_call(self, operation_name, api_params):
     )
 
 
-@pytest.mark.django_db
 def test_upload_archive_one_uploaded_one_skipped(
     client,
-    botomock,
-    fakeuser,
+    db,
+    s3_helper,
+    tmp_path,
+    uploaderuser,
 ):
-    token = Token.objects.create(user=fakeuser)
+    token = Token.objects.create(user=uploaderuser)
     (permission,) = Permission.objects.filter(codename="upload_symbols")
     token.permissions.add(permission)
-    url = reverse("upload:upload_archive")
-
-    def mock_api_call(self, operation_name, api_params):
-        # This comes for the setting UPLOAD_DEFAULT_URL specifically
-        # for tests.
-        assert api_params["Bucket"] == "publicbucket"
-
-        if operation_name == "HeadBucket":
-            # yep, bucket exists
-            return {}
-
-        if operation_name == "HeadObject" and api_params["Key"] == (
-            "v0/flag/deadbeef/flag.jpeg"
-        ):
-            # file exists with same size
-            return {"ContentLength": 69183}
-
-        if operation_name == "HeadObject" and api_params["Key"] == (
-            "v0/xpcshell.dbg/A7D6F1BB18CD4CB48/xpcshell.sym"
-        ):
-            # Not found at all
-            parsed_response = {"Error": {"Code": "404", "Message": "Not found"}}
-            raise ClientError(parsed_response, operation_name)
-
-        if operation_name == "PutObject" and api_params["Key"] == (
-            "v0/xpcshell.dbg/A7D6F1BB18CD4CB48/xpcshell.sym"
-        ):
-            # Successfully uploaded
-            return {}
-
-        raise NotImplementedError((operation_name, api_params))
+    s3_helper.create_bucket("publicbucket")
+
+    # Upload flag.jpeg from the zip file into the bucket so it's already there
+    # and gets ignored when it's uploaded
+    rootdir = str(tmp_path)
+    with open(ZIP_FILE, "rb") as fp:
```

Comment:
I had to pull the file out of the ZIP_FILE to upload it so that it was already in S3. This is weird. Instead of having pre-built zip files which are opaque and harder to reason about in the tests ("wtf is in _this_ zip file? i have no idea when reading this test."), we should build the zip files with the contents we need them to have. Then we wouldn't have to do this weird thing where we extract the file from the zip file to upload it.

Also, we'd be able to more easily create well-formed files--the debug id in one or more of the zip files is invalid.